templates/ceph-storage.yaml.j2

```diff
@@ -0,0 +1,256 @@
+      Unless the default is changed, the parameter is automatically resolved
+      from the subnet host_routes attribute.
+    type: json
+  ManagementInterfaceRoutes:
+    default: []
+    description: >
+      Routes for the management network traffic.
+      JSON route e.g. [{'destination':'10.0.0.0/16', 'nexthop':'10.0.0.1'}]
+      Unless the default is changed, the parameter is automatically resolved
+      from the subnet host_routes attribute.
+    type: json
+{%raw%}{% endif %}{%endraw%}
+resources:
+  OsNetConfigImpl:
+    type: OS::Heat::SoftwareConfig
+    properties:
+      group: script
+      config:
+{% if public_external_interface is defined %}
+        list_join:
+        - ''
+        - - |
+            #!/bin/bash
+            for f in $(find /proc/sys/net/ipv4 -name rp_filter) ; do echo 0 > $f ; done
+
+          -
+{% endif %}
+            str_replace:
+              template:
+                  get_file: {%raw%}{{ install.heat.templates.basedir }}{%endraw%}/network/scripts/run-os-net-config.sh
+
+              params:
+                   $network_config:
+                       network_config:
+                        -
+                          type: ovs_bridge
+                          name: br-ex
+                          use_dhcp: false
+{% if external_interface == ctlplane_interface %}
+                          addresses:
+                            -
+                              ip_netmask:
+                                list_join:
+                                  - '/'
+                                  - - get_param: ControlPlaneIp
+                                    - get_param: ControlPlaneSubnetCidr
+                          routes:
+                            -
+                              ip_netmask: 169.254.169.254/32
+                              next_hop:
+                                  get_param: EC2MetadataIp
+{% endif %}
+                          members:
+                            -
+                              type: interface
+                              name: {{ external_interface }}
+                            -
+                              type: vlan
+                              vlan_id:
+                                  get_param: StorageMgmtNetworkVlanID
+                              addresses:
+                            - 
+                              ip_netmask:
+                                  get_param: StorageMgmtIpSubnet
+
+
+{% if storage_interface is defined %}
+                        - 
+                          type: ovs_bridge
+                          name: br-storage
+                          use_dhcp: false
+                          members:
+                            - 
+                              type: interface
+                              name: {{ storage_interface }}
+{% endif %}
+
+                            - 
+                              type: vlan
+                              vlan_id:
```

Comment:
I'm not sure which of these are really necessary to add to your PR, but here's an example of what older OSP-Ceph templates (from Dec. 2016 = RHOSP 13?)  had to add, will be looking to see if your PR generates equivalent templates.

in [example of network-environment.yaml from older OSP](http://perf1.perf.lab.eng.bos.redhat.com/pub/bengland/public/openstack/templates/network-environment.yaml) : 

```
resource_registry:
...
  OS::TripleO::CephStorage::Net::SoftwareConfig: /home/stack/templates/nic-configs/ceph-storage.yaml
...
parameter_defaults:
  StorageMgmtNetCidr: 172.19.0.0/24
  StorageMgmtAllocationPools: [{'start': '172.19.0.10', 'end': '172.19.0.200'}]
  StorageMgmtNetworkVlanID: 203
...
```

in [older storage-environment.yaml](http://perf1.perf.lab.eng.bos.redhat.com/pub/bengland/public/openstack/templates/storage-environment.yaml) :

```
resource_registry:
  OS::TripleO::NodeUserData: /home/stack/wipe-disks.yaml
  OS::TripleO::Services::CephMon: /usr/share/openstack-tripleo-heat-templates/puppet/services/ceph-mon.yaml
  OS::TripleO::Services::CephOSD: /usr/share/openstack-tripleo-heat-templates/puppet/services/ceph-osd.yaml
  OS::TripleO::Services::CephClient: /usr/share/openstack-tripleo-heat-templates/puppet/services/ceph-client.yaml
...
parameter_defaults:
  CinderEnableRbdBackend: true
  NovaEnableRbdBackend: true
  GlanceBackend: rbd
...
 ExtraConfig:
    ceph::profile::params::osds:
      '/dev/sdb':
        journal: '/dev/nvme0n1'
      '/dev/sdc':
        journal: '/dev/nvme0n1'
...
```
in [nic-configs/controller.yaml](http://perf1.perf.lab.eng.bos.redhat.com/pub/bengland/public/openstack/templates/nic-configs/controller.yaml) :

```
resources:
  OsNetConfigImpl:
    type: OS::Heat::StructuredConfig
    properties:
      group: os-apply-config
      config:
        os_net_config:
          network_config:
...
            -
              type: ovs_bridge
              name: br-storage
              use_dhcp: false
              dns_servers: {get_param: DnsServers}
              members:
                 -
                   type: interface
                   name: nic4
                 -
                   type: vlan
                   device: br-storage
                   vlan_id: {get_param: StorageNetworkVlanID}
                   addresses:
                   -
                    ip_netmask: {get_param: StorageIpSubnet}
                 -
                   type: vlan
                   device: br-storage
                   vlan_id: {get_param: StorageMgmtNetworkVlanID}
                   addresses:
                   -
                    ip_netmask: {get_param: StorageMgmtIpSubnet}
 ```
 in [nic-configs/compute.yaml](http://perf1.perf.lab.eng.bos.redhat.com/pub/bengland/public/openstack/templates/nic-configs/compute.yaml.tim) : 

```
resources:
  OsNetConfigImpl:
    type: OS::Heat::StructuredConfig
    properties:
      group: os-apply-config
      config:
        os_net_config:
          network_config:
...
                    ip_netmask: {get_param: InternalApiIpSubnet}
            -
              type: ovs_bridge
              name: br-storage
              use_dhcp: false
              dns_servers: {get_param: DnsServers}
              members:
                 -
                   type: interface
                   name: nic4
                 -
                   type: vlan
                   device: br-storage
                   vlan_id: {get_param: StorageNetworkVlanID}
                   addresses:
                   -
                    ip_netmask: {get_param: StorageIpSubnet}
                # Uncomment when including environments/network-management.yaml
                # If setting default route on the Management interface, comment
                # out the default route on the Control Plane.
                #-
                #  type: vlan
                #  vlan_id: {get_param: ManagementNetworkVlanID}
                #  addresses:
                #    -
                #      ip_netmask: {get_param: ManagementIpSubnet}
                #  routes:
                #    -
                #      default: true
                #      next_hop: {get_param: ManagementInterfaceDefaultRoute}
```
because compute nodes don't need access to the StorageManagement network (used for Ceph replication).